tests/test_integration.py

```diff
@@ -424,6 +428,22 @@ def test_info(self):
         self.vm.shutdown()
         host_info(VM_HOSTNAME)
 
+    def test_vm_define(self):
+        vm_dataset_obj = Query({'hostname': VM_HOSTNAME}, VM_ATTRIBUTES).get()
+        vm = VM(vm_dataset_obj)
+
+        hv_dataset_obj = Query(
+            {'hostname': vm_dataset_obj['hypervisor']},
+            HYPERVISOR_ATTRIBUTES).get()
+        hv = Hypervisor(hv_dataset_obj)
+
+        vm_stop(VM_HOSTNAME)
+        hv.undefine_vm(vm, keep_storage=True)
+
+        self.assertFalse(hv.vm_defined(vm))
+        vm_define(VM_HOSTNAME)
+        self.assertTrue(hv.vm_defined(hv.vm_defined(vm)))
```

Comment:
this call looks weird. is it correct?

Reply:
we can rather use         self.check_vm_present()

Reply:
Yes, I wanted to have the tests passed first but they did not start.